### PR TITLE
Lazy alerts

### PR DESCRIPTION
**Pull request by DenverM80**

### CODE REVIEW

**Comment by rpmoore** [comment]
out of date

**Comment by DenverM80** [comment]
Updated

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java [hunk 1]
```diff
@@ -158,7 +160,7 @@ public void initialize(final URL location, final ResourceBundle resources) {
             ds3Common.setDeepStorageBrowserPresenter(deepStorageBrowserPresenter);
             try {
                 //open default session when DSB launched
-                savedSessionStore.openDefaultSession(ds3SessionStore);
+                savedSessionStore.openDefaultSession(ds3SessionStore, resourceBundle);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java]: Why is the resource bundle needed here?  Why is it  not injected into the savedSessionStore when it is created?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java [hunk 2]
```diff
@@ -546,22 +548,22 @@ public void ds3DeleteObject(final boolean isButtonEvent) {
         if (Guard.isNullOrEmpty(values)) {
             if (root.getValue() == null) {
                 LOG.info("No files selected");
-                Ds3Alert.show(resourceBundle.getString("error"), resourceBundle.getString("noFiles"), Alert.AlertType.ERROR);
+                alert.showAlert(resourceBundle.getString("noFiles"));
             } else {
                 final ImmutableList.Builder<TreeItem<Ds3TreeTableValue>> builder = ImmutableList.builder();
-                values = builder.add(root).build().asList();
+                values = builder.add(root).build().asList();  //DM TODO why?
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java]: If values is only being read, remove the `.asList()`

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/ds3Panel/CreateService.java [hunk 3]
```diff
@@ -28,11 +26,14 @@
 
 public final class CreateService {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/ds3Panel/CreateService.java]: Not in this pull request, but we need to turn this into a real service where any of the internal dependencies get injected, and more importantly, where all the ui logic is removed.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+
+package com.spectralogic.dsbrowser.gui.util;
+
+import javafx.application.Platform;
+import javafx.scene.control.Alert;
+import javafx.scene.image.Image;
+import javafx.stage.Stage;
+
+public class LazyAlert {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/LazyAlert.java]: Add some javadoc explaining what makes this lazy.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java (hunk 2 above)
Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java]: values is never used.  I think the else can be removed.

Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java]: Looking closer at it, I think you are right.  Go ahead and remove it and see what happens.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java [hunk 5]
```diff
@@ -546,22 +548,22 @@ public void ds3DeleteObject(final boolean isButtonEvent) {
...
             }
         } else if (values.stream().map(TreeItem::getValue).anyMatch(value -> value.getType() == Ds3TreeTableValue.Type.Directory)) {
             if (isButtonEvent) {
```
Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java]: @rpmoore Is there any reason this should behave differently if its a ButtonEvent or from a right-click menu?  I can't think of any.  

The current behavior is that the Button event deletes the folder along with the item if its the only item in the folder.  The right-click menu correctly only deletes the item.

**Comment by DenverM80** [comment]
updated

**Pull request merged**